### PR TITLE
fix(datasets) Fix the way split query works on joined dataset

### PR DESCRIPTION
**Pull request by fpacifici**
PR #488 moved the list of columns to use to perform query split by column into the dataset.
This did not work for joined datasets because:
1) the methods were not added to the groups dataset
2) split.py would still have the column names hardcoded, thus it would miss the table alias (this issue was not introduced by the PR above).

Fixing that by turning get_min_columns and get_timestamp_column into a ColumnSplitSpec object that can be None. When it is none split.py does not perform column split.

test plan:
- test_api.py has a test for query column split. It still works.
- sentry test-snuba work
- this query works:

```
{
    "selected_columns": ["events.event_id", "events.tags[level]", "events.timestamp", "events.tags[transaction]"],
    "project": [2],
    "from_date": "2019-09-26T19:53:39",
    "to_date": "2019-10-02T19:53:39",
    "granularity": 3600
}

{
    "data": [
        {
            "events.event_id": "e7cab87ad7ae4d49b91e5ab3328a845b",
            "events.tags[level]": "error",
            "events.timestamp": "2019-10-01T22:33:36+00:00",
            "events.tags[transaction]": null
        },
        {
            "events.event_id": "d485a02c165b40189fadf3258fabc4a2",
            "events.tags[level]": "error",
            "events.timestamp": "2019-10-01T22:33:42+00:00",
            "events.tags[transaction]": null
        },
        {
            "events.event_id": "4ca26033c8ee43f79899f7a745bedeed",
            "events.tags[level]": "error",
            "events.timestamp": "2019-10-01T22:33:42+00:00",
            "events.tags[transaction]": null
        },
        {
            "events.event_id": "57c27b172f7644feafaad5cd53a5fcce",
            "events.tags[level]": "error",
            "events.timestamp": "2019-10-01T22:33:39+00:00",
            "events.tags[transaction]": null
        }
    ],
```

### CODE REVIEW

File: snuba/split.py [hunk 1]
```diff
@@ -133,8 +138,11 @@ def col_split(dataset, request: Request, *args, **kwargs):
         # The query function may mutate the request body during query
         # evaluation, so we need to copy the body to ensure that the query has
         # not been modified by the time we're ready to run the full query.
+        column_split_spec = dataset.get_split_query_spec()
```
[the remark this answers is not in the document]
Comment (fpacifici) [reply snuba/split.py]: Ok, makes sense, will do.

**Comment by fpacifici** [comment]
Added a test for column split and embedded Manu's comment

File: tests/test_split.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    def do_query(dataset, request: Request, timer):
+        assert request.query == query
+
+    request = Request(
+        query,
+        RequestSettings(False, False, False),
+        {},
+    )
+
+    events = get_dataset(dataset)
```
Comment (manuzope) [inline tests/test_split.py]: Maybe
```
dataset = get_dataset(dataset_name)
```

File: tests/test_split.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        "selected_columns": ["event_id"],
+        "conditions": [""],
+        "orderby": "event_id",
+        "sample": 10,
+        "limit": 100,
+        "offset": 50,
+    })
+
+    @split_query
+    def do_query(dataset, request: Request, timer):
```
Comment (manuzope) [inline tests/test_split.py]: Just curious: is there a reason for putting types on some of the params and not all?

Comment (fpacifici) [reply tests/test_split.py]: fixed.

File: tests/test_split.py (hunk 2 above)
Comment (fpacifici) [reply tests/test_split.py]: fixed.

**Pull request merged**